lib/rules/match-component-file-name.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * @fileoverview Require component name property to match its file name
+ * @author Rodrigo Pedra Brum <rodrigo.pedra@gmail.com>
+ */
+'use strict'
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const utils = require('../utils')
+const path = require('path')
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'require component name property to match its file name',
+      category: 'recommended',
+      url: 'https://github.com/vuejs/eslint-plugin-vue/blob/v5.0.0-beta.4/docs/rules/match-component-file-name.md'
+    },
+    fixable: null,
+    schema: [
+      { enum: ['jsx', 'both'] }
+    ]
+  },
+
+  create (context) {
+    return utils.executeOnVueComponent(context, (object) => {
+      const nameProperty = object.properties.find((prop) => prop.key.name === 'name')
```

Comment:
spread operator is going to crash here

```suggestion
      const nameProperty = object.properties
        .find(p =>
          p.type === 'Property' &&
          p.key.type === 'Identifier' &&
          p.key.name === 'name' &&
          (
            p.value.type === 'Literal' || (
              p.value.type === 'TemplateLiteral' &&
              p.value.expressions.length === 0 &&
              p.value.quasis.length === 1
            )
          )
        )
```

Reply:
i didn't tested it, i just wrote it ~ how i remember AST

Reply:
I used and adapted version from your code in [component-definition-name-casing](https://github.com/vuejs/eslint-plugin-vue/pull/646/files#diff-14393fb5384e4c5a97aa80b9c06923b2R74) rule